File: android_app/app/src/main/java/com/oceansentinels/app/domain/repository/AuthRepository.kt
package com.oceansentinels.app.domain.repository

import com.oceansentinels.app.domain.model.*
import kotlinx.coroutines.flow.Flow

/**
 * Repository interface for authentication operations
 */
interface AuthRepository {
    
    /**
     * Get the current user session as a flow
     */
    fun getUserSession(): Flow<UserSession>
    
    /**
     * Login with username and password
     */
    suspend fun login(credentials: LoginCredentials): Result<AuthToken>
    
    /**
     * Register a new user
     */
    suspend fun register(data: RegistrationData): Result<User>
    
    /**
     * Logout the current user
     */
    suspend fun logout()
    
    /**
     * Get the current authenticated user
     */
    suspend fun getCurrentUser(): Result<User>
    
    /**
     * Check if user is authenticated
     */
    suspend fun isAuthenticated(): Boolean
    
    /**
     * Get stored auth token
     */
    suspend fun getStoredToken(): String?
    
    /**
     * Refresh the auth token
     */
    suspend fun refreshToken(): Result<AuthToken>
}

File: IOS/OceanSentinels/Domain/Repositories/AuthRepository.swift
import Foundation

/// Repository protocol for authentication operations.
protocol AuthRepository {

    /// Authenticates a user with the given credentials.
    func login(credentials: LoginCredentials) async throws -> AuthToken

    /// Registers a new user account.
    func register(data: RegistrationData) async throws -> User

    /// Logs out the current user, clearing stored session data.
    func logout() async

    /// Checks whether the current user is authenticated.
    func isAuthenticated() -> Bool

    /// Returns the current user session state.
    func getUserSession() -> UserSession

    /// Retrieves the stored authentication token, if available.
    func getStoredToken() -> String?

    /// Fetches the currently authenticated user's profile.
    func getCurrentUser() async throws -> User

    /// Restore auth state on app launch.
    func restoreSession() async
}
